app/concepts/tickets/use_cases/generate_qr_code.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Tickets
+  module UseCases
+    class GenerateQrCode
+      def call(ticket_id)
+        {
+          id: ticket_id,
+          key: Ticket.find_by(id: ticket_id).key
```

Comment:
```suggestion
          key: Ticket.find(ticket_id).key
```

`Ticket.find_by(id: ...)` returns `nil` if no ticker with specified `id` was found, so then `.key` will raise here following error:
```
undefined method `key' for nil:NilClass (NoMethodError)
```
Which then would be transalted to 500 in production mode.

`Ticket.find(...)` will raise `ActiveRecord::RecordNotFound` which should be default in production be translated to 404 error, which would be better here.

Reply:
O, and maybe let's use [repository here?](https://github.com/Joannitka-pl/monte-cinema/blob/8a11a4c2ede719d3fceb4b6d74869b41b251d000/lib/base_repository.rb)